### PR TITLE
Feat/notis

### PR DESCRIPTION
**Pull request by TianYueDong**
Briefly describe the user story, for example:

As a project owner, I would like to have notifications so I can be informed when a user requests to join my project.
As a project member, I would like to have notifications so I can be informed when I've been invited to a project.

Acceptance Criteria
Paginated API route GET /api/v1/notifications/ for fetching notifications
API route DELETE /api/v1/notifications/{noti_id} for deleting notification (click read in frontend).

### CODE REVIEW

File: Buddies.API/Controllers/NotificationsController.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                    sb.Append("yesterday");
+                }
+                else if (ts.TotalDays < 28)
+                {
+                    sb.AppendFormat("{0} weeks ago", (int)(ts.Days / 7));
+                }
+                else 
+                {
+                    sb.AppendFormat("{0} months ago", (int)(ts.Days / 30));
+                }
```
Comment (andy9775) [inline Buddies.API/Controllers/NotificationsController.cs:76]: Is there a third party library we could use for this type of thing?

File: Buddies.Web/src/pages/notifications.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import CardContent from '@mui/material/CardContent';
+import CardActionArea from '@mui/material/CardActionArea';
+
+// const getapi = axios.create({
+//     baseURL: '/api/v1/notifications/',
+//     headers: {
+//         Accept: 'application/json',
+//         'Content-type': 'application/json',
+//     },
+// });
```
Comment (andy9775) [inline Buddies.Web/src/pages/notifications.tsx]:
```suggestion
```

File: Buddies.Web/src/pages/notifications.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  'projectId': number,
+  'isRead': boolean,
+  'timeCreated': string
+};
+
+const Notifications = () => {
+  let currentPage = 1;
+  const size = 20;
+  const [notis, setNotis] = useState<NotiObject[]>([]);
+  console.log(notis);
```
Comment (andy9775) [inline Buddies.Web/src/pages/notifications.tsx]:
```suggestion
```

File: Buddies.Web/src/pages/notifications.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        data.notifications.forEach((p: NotiObject) => currNoti.push(p));
+        setNotis((noti) => [...noti, ...currNoti]);
+      })
+      .catch((error) => {
+        alert(error);
+      });
+    currentPage += 1;
+  };
+
+  const handleScroll = (e:any) => {
```
Comment (andy9775) [inline Buddies.Web/src/pages/notifications.tsx]: any?

File: Buddies.Web/src/pages/notifications.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  'isRead': boolean,
+  'timeCreated': string
+};
+
+const Notifications = () => {
+  let currentPage = 1;
+  const size = 20;
+  const [notis, setNotis] = useState<NotiObject[]>([]);
+  console.log(notis);
+  const [bot, setBot] = useState<boolean>(false);
```
Comment (andy9775) [inline Buddies.Web/src/pages/notifications.tsx]: More descriptive variable name?

File: Buddies.Web/src/pages/projects/[pid].tsx [hunk 6]
```diff
@@ -111,6 +112,19 @@ const Project: React.VFC = () => {
       .catch((err) => displayErrorNotif(err));
   };
 
+  const requestToJoin = async () => {
+    if (!(typeof projectId === 'string')) {
+      alert('error');
+      return;
+    }
+    const res = await joinRequest(projectId)
+      .catch((error) => alert(error));
```
Comment (andy9775) [inline Buddies.Web/src/pages/projects/[pid].tsx]: The `alert` calls aren't very user friendly. Perhaps just show a toast with a generic error, or do something on behalf of the user?

**Review by andy9775 — APPROVED** [review]
Just some minor nits and comments, and there look to be some warnings from the tests, make sure you review them.

Otherwise LGTM!

**Review by stalhaali — APPROVED** [review]
LGTM!

**Pull request merged**